### ISSUE TITLE
ARAX_filter_results

### ISSUE THREAD

**Issue by dkoslicki** [RTXteam/RTX#629]
We will need the ability to filter results. @finnagin, similar to how you implemented ARAX_filter_kg, we will need an `ARAX_filter_results`. Of course, you won't be able to actually work on this until after @saramsey finishes the `recompute_results()` and @edeutsch writes the wrapper (see #618).

For the demo, the requirements are at minimum the following DSL command
```
filter_results(action=sort_by_edge_attribute, edge_attribute=jaccard_index, edge_type=J1, direction=descending, max_n_results=10)
```
Other possible `action`s include (in decreasing order of priority):
- [x] sort_by_edge_attribute (I assume here you may need to make an optional allowable_parameter to make the sorting based off the max or the total if a result contains more than one edge with the indicated attribute. edge_type should also be optional/not required)
- [x] sort_by_node_attribute (same proviso re: max/total if more than one node has the indicated attribute)
- [x] limit_number_of results (i.e. just return 10 of them, I don't care how they are sorted)
- [x] sort_by_node_count
- [x] sort_by_edge_count
- [ ] remove_virtual_edges

In each action it makes sense to, you will probably want to include `allowable_parameters`:
- [x] direction=
- [x] max_n_results=

May want to also include option to filter results based on their similarity, so something like:
- [ ] action=filter_by_similarity, similarity_type={#nodes in common, # edges in common, #specific drug class, etc. etc.)

**Comment by dkoslicki** [comment]
@finnagin now that #618 is working somewhat (ping me if you want to know how to use it). I think we should **really** push on this one. Let me know your bandwidth and if I should focus on this 2/21 (tomorrow) or if you'll be able to handle it.

**Comment by dkoslicki** [comment]
@finnagin Note, I merged `overlay` into `filter`, so you should be good to go right away

**Comment by dkoslicki** [comment]
@finnagin on overlay, example 17 returns results

**Comment by finnagin** [comment]
@dkoslicki I've added filter results code with the following actions: 'sort_by_edge_attribute',  'sort_by_node_attribute', 'limit_number_of_results', 'sort_by_edge_count', and 'sort_by_node_count'.

While I've written the code it still needs to be tested. I'll be able to test it after #653 is resolved.

**Comment by dkoslicki** [comment]
Now that #653 is resolved, feel free to check off your progress in the top comment of this issue. That way, we (and @edeutsch) can know exactly what DSL commands to include in the demo iPython notebooks! Thanks for implementing this so quickly!

Friendly reminder: Make sure that the DSL documentation stuff is there and re-run the `RTX/code/ARAX/Documentation/document_dsl_commands.py` and make sure the resulting `RTX/code/ARAX/Documentation/DSL_Documentation.md` is to your liking!

**Comment by finnagin** [comment]
@finnagin Reminder to myself that I need to update the documentation for the results query functions

**Comment by dkoslicki** [comment]
@finnagin just FYI, if/when you get to `remove_virtual_edges`, you’ll need to make sure to remove them from the query graph as well (to keep everything API standards compliant).

Also, feel free to modify the [ARX DSL Examples](https://docs.google.com/document/d/1VweSOVxZS-7rXMc2ZhaZ6fpqYKaxRH_Vht0-PXRmy84) to include your new filter commands in the demo examples 2 and 3 (in example 2, it’s highlighted in red where you can put your new DSL commands).

**Comment by dkoslicki** [comment]
@finnagin I think this is mostly complete.
The item:
>May want to also include option to filter results based on their similarity, so something like:
> action=filter_by_similarity, similarity_type={#nodes in common, # edges in common, #specific drug class, etc. etc.)

Can be addressed later if needed.

Any thoughts about:
> remove_virtual_edges

This might be difficult to do in general if others are making virtual edges. But your thoughts on this would be nice before we close this issue.

**Comment by finnagin** [comment]
@dkoslicki a couple ways we could implement `remove_virtual_edges` either have a flag or prefix in the type or something identifying it as a virtual edge or we could have a hardcoded list of all the virtual edge types we use and remove them that way. Thoughts about those options?

**Comment by dkoslicki** [comment]
@finnagin sorry this fell off my radar. At this point, I think removing virtual edges is low priority until someone objects. I’ll close this issue as most of the functionality is already implemented and other work is ongoing #788 #785
